### PR TITLE
Implement mm_max_epi8

### PR DESCRIPTION
**Pull request by HowJMay**

### CODE REVIEW

File: sse2neon.h [hunk 1]
```diff
@@ -2862,6 +2862,14 @@ FORCE_INLINE __m128i _mm_min_epi16(__m128i a, __m128i b)
         vminq_s16(vreinterpretq_s16_m128i(a), vreinterpretq_s16_m128i(b)));
 }
 
+// Computes the pairwise maxima of the 16 signed 8-bit integers from a and the
+// 16 signed 8-bit integers from b.
+FORCE_INLINE __m128i _mm_max_epi8(__m128i a, __m128i b)
```
Comment (jserv) [inline sse2neon.h:2867]: Add hyperlink to Intel website.

File: tests/impl.h [hunk 2]
```diff
@@ -138,6 +138,7 @@
     TYPE(mm_loadu_pd)                 \
     TYPE(mm_loadu_si128)              \
     TYPE(mm_madd_epi16)               \
+    TYPE(mm_max_epi8)                 \
```
Comment (jserv) [inline tests/impl.h:141]: `_mm_max_epi8` came with SSE4.1 and you shall move the item into the classified area.

File: sse2neon.h (hunk 1 above)
Comment (marktwtn) [reply sse2neon.h:2867]: Use the description of the hyperlink as well.

File: sse2neon.h [hunk 3]
```diff
@@ -2862,6 +2862,14 @@ FORCE_INLINE __m128i _mm_min_epi16(__m128i a, __m128i b)
         vminq_s16(vreinterpretq_s16_m128i(a), vreinterpretq_s16_m128i(b)));
 }
 
+// Compare packed signed 8-bit integers in a and b, and store packed maximum values in dst.
+// https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=max_epi8&expand=3594
```
Comment (marktwtn) [inline sse2neon.h:2866]: Replace with `https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm_max_epi8`.

**Review by marktwtn — APPROVED** [review]

**Comment by marktwtn** [comment]
Check the coding convention.

**Comment by jserv** [comment]
> Check the coding convention.

@HowJMay, execute `make indent` to ensure the consistent style.

**Pull request merged**